src/privileged/fxa/api.js

```diff
@@ -11,20 +11,40 @@ ChromeUtils.import("resource://gre/modules/FxAccountsCommon.js");
 ChromeUtils.defineModuleGetter(this, "fxAccounts", "resource://gre/modules/FxAccounts.jsm");
 ChromeUtils.defineModuleGetter(this, "EnsureFxAccountsWebChannel", "resource://gre/modules/FxAccountsWebChannel.jsm");
 ChromeUtils.defineModuleGetter(this, "BrowserWindowTracker", "resource:///modules/BrowserWindowTracker.jsm");
+ChromeUtils.defineModuleGetter(this, "Weave", "resource://services-sync/main.js");
 
 /* eslint-disable no-undef */
 const { EventManager } = ExtensionCommon;
 const EventEmitter = ExtensionCommon.EventEmitter || ExtensionUtils.EventEmitter;
 
 function sanitizeUser(user) {
-  if (user && user.profileCache && user.profileCache.profile) {
+  if (user) {
+    let avatar, email;
+    const { verified } = user;
+
+    if (user.profileCache && user.profileCache.profile) {
+      avatar = user.profileCache.profile.avatar;
+      email = user.profileCache.profile.email;
+    }
+
     return {
-      ...user.profileCache.profile,
-      verified: user.verified,
+      avatar,
+      email,
+      hashedUid: hashedUid(),
+      verified,
     };
   }
 
-  return user;
+  return undefined;
```

Comment:
Figured I'd be explicit here for anyone who is just scanning the code, this function will not always return a valid object.